cmd/experimental/kueue-viz/frontend/Dockerfile

```diff
@@ -1,19 +1,38 @@
-#Build the React app
-FROM node:23 AS build
+# Build stage
+FROM node:23 AS builder
 
-ENV NPM_CONFIG_CACHE=/tmp/.npm-cache
+# Set working directory
 WORKDIR /app
+
+# Copy package files
 COPY package*.json ./
+
+# Install ALL dependencies (including dev dependencies needed for build)
 RUN npm install
```

Comment:
Do we need dev dependencies?

Reply:
Usually, dev dependencies contains utils needed for minification for example. This line is for the builder image, so it makes sense to have the dev dependencies here.

Reply:
The purpose of dev dependencies is to avoid installing libraries needed only for development. If a dependency is required for the build process, I think it should be listed under dependencies.